### ISSUE TITLE
Tabbar Visibility parameter missing when using VisualStateManger c#

### ISSUE THREAD

**Issue by MarkLFT** [unoplatform/uno.toolkit.ui#1105]
## Current behavior
I have created a page that uses c# for defining the layout page.  When I add a tabbar, I can set the property for visibility to collapsed or visible.  However, in the VisualStateManager, defined in the parent grid, when I try to set the visibility using a setter, I receive a red underline.  

## Expected behavior

I expect all the properties available in the tabbar definition to be available as a setter in the visual state manager.

## How to reproduce it (as minimally and precisely as possible)

The Tabbar is defined as below, within a Grid.

```
new TabBar()
    .Grid(row: 1)
    .Region(attached: true)
    .Name("tabBar")
    .Assign(out TabBar tabBar)
    .Style(Uno.Toolkit.UI.Markup.Theme.TabBar.Styles.Bottom)
    .VerticalAlignment(VerticalAlignment.Bottom)
    .Items(
        new TabBarItem()
            .Icon(new SymbolIcon(Symbol.Home))
            .Region(name: "home"),
        new TabBarItem()
            .Icon(new SymbolIcon(Symbol.Contact))
            .Region(name: "rates"),
        new TabBarItem()
            .Icon(new SymbolIcon(Symbol.Shop)))
            .Region(name: "settings")
```

Within the same grid I define the following VisualStateManager

```
.VisualStateManager(vsm =>
    vsm.Group("vsm_group", grid =>
        grid.State(name: "Narrow", state => state
                .Setters(tabBar, tb => tb.Visibility(Visibility.Visible))
                .Setters(navView, nv => {
                    nv.IsPaneToggleButtonVisible(false);
                    nv.PaneDisplayMode(NavigationViewPaneDisplayMode.LeftMinimal);
                    nv.IsPaneOpen(false);
                }),
        grid.State(name: "Normal", state => state
                .StateTriggers(new AdaptiveTrigger().MinWindowWidth(700))
                .Setters(tabBar, tb => tb.Visibility(Visibility.Visible))
                .Setters(navView, nv => {
                    nv.IsPaneToggleButtonVisible(true);
                    nv.PaneDisplayMode(NavigationViewPaneDisplayMode.Auto);
                    nv.IsPaneOpen(true);
                }),
        grid.State(name: "WideStateTrigger", state => state
                .StateTriggers(new AdaptiveTrigger().MinWindowWidth(1000))
                .Setters(tabBar, tb => tb.Visibility(Visibility.Collapsed)
                .Setters(navView, nv => {
                    nv.IsPaneToggleButtonVisible(true);
                    nv.PaneDisplayMode(NavigationViewPaneDisplayMode.Auto);
                    nv.IsPaneOpen(true);
                })
            )
        )
)
```

If I place a .Visibility in the Tabbar definition I get no errors, however in the Setters the tb.Visibility is underlined in red as not a valid property.

Nuget Package: 5.1.7

Package Version(s):
Uno.Platform  5.1.104
Uno.WinUI.Markup 5.1.5
Uno.Toolkit.WinUI.Material.Markup  5.1.7

Affected platform(s):

- [x ] WebAssembly
- [x ] Android
- [x ] iOS
- [x ] macOS (AppKit)
- [x ] Mac Catalyst
- [x ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Linux Framebuffer
  - [ ] Tizen
- [x ] Windows

IDE:

- [x ] Visual Studio 2022
- [ ] Visual Studio 2019
- [ ] Visual Studio Code
- [ ] Visual Studio for Mac
- [ ] Rider Windows
- [ ] Rider macOS

Relevant plugins: None

**Comment by kazo0** [comment]
Thanks @MarkLFT for the report!

What is the error that you see in the logs if you try and build?

**Comment by MarkLFT** [comment]
@kazo0, Thanks for your prompt response.

The build error message is:

The type 'Uno.Extensions.Markup.StyleBuilder<Uno.Toolkit.UI.TabBar>' cannot be used as type parameter 'T' in the generic type or method 'UIElementMarkup.Visibility<T>(T, Visibility)'. There is no implicit reference conversion from 'Uno.Extensions.Markup.StyleBuilder<Uno.Toolkit.UI.TabBar>' to 'Microsoft.UI.Xaml.UIElement'

In addition, the intellisense does nto show the parameter for Visibility, whereas it does show the intellisense for the NavigationView setters.

**Comment by kazo0** [comment]
@dansiegel ring any bells? Will look into it and see if I can find anything

**Comment by kazo0** [comment]
@MarkLFT nice catch :) 

@eriklimakc is still investigating and we will have a fix coming soon. Stay tuned as we will try and see if there's some sort of workaround we can come up with to unblock you for now

**Comment by dansiegel** [comment]
Fixed in:
- unoplatform/uno.csharpmarkup#698
- unoplatform/uno.csharpmarkup#702

This is currently available in 5.3.0-dev.36 and will be available in the 5.2 release
